kernel/Kconfig

```diff
@@ -813,6 +813,7 @@ config SYS_CLOCK_TICKS_PER_SEC
 
 config SYS_CLOCK_HW_CYCLES_PER_SEC
 	int "System clock's h/w timer frequency"
+	default 0 if TIMER_READS_ITS_FREQUENCY_AT_RUNTIME
```

Comment:
There's a risk that changes like this can break out-of-tree/app code that was making bad assumptions and failing to use `sys_clock_hw_cycles_per_sec()`, though I'm not entirely sure that's a bad thing.